scRNA-seq-advanced/01-reading_filtering_CellRanger.Rmd

```diff
@@ -0,0 +1,484 @@
+
+
+Now we can filter our `raw_sce` object _by column_ to only keep the cells with a small FDR: those that are quite unlikely to be empty droplets. 
+
+```{r filter emptydrops}
+cells_to_retain <- which(droplet_df$FDR < 0.01)
+filtered_sce <- raw_sce[, cells_to_retain]
+```
+
+How many cells do we have now? 
+
+```{r filtered summary}
+filtered_sce
+```
+
+## Additional Quality control
+
+In addition to filtering out empty droplets, we also will want to filter out real cells that may have been damaged during preparation of the library.
+These will often be characterized by a high proportion of mitochondrial transcripts.
+The idea here is that if a cell ruptures, the cytoplasmic transcripts will leak out, but mitochondrial transcripts, protected by the mitochondrial membrane, will be more protected. 
+
+Our first step then, is create a list of the mitochondrial genes that are present in our dataset.
+
+```{r get mitochondrial genes}
+# read in a table of mitochondrial genes
+mito_genes <- readr::read_tsv(mito_file) %>%
+  # filter to only the genes that are found in our dataset
+  dplyr::filter(gene_id %in% rownames(filtered_sce)) %>%
+  # create a vector from the gene_id column
+  dplyr::pull(gene_id)
+```
+
+### Calculating summary QC statistics 
+
+We can now use the `scuttle` function `addPerCellQC()` to calculate some statistics based on the counts matrix, which will be added to the `colData` data frame.
+
+In addition to calculating statistics like the total read count for each cell and the number of transcripts that are detected, we can also calculate those statistics for defined subsets of genes. 
+In this case, we will use our `mito_genes` vector to define a subset called `mito`.
+The `mito` name is important in that it is the name that will be expected by a later function.
+(We could define more subsets, but for now this one will do.)
+
+```{r per cell QC}
+filtered_sce <- scuttle::addPerCellQC(filtered_sce,
+                                      subsets = list(mito = mito_genes))
+```
+
+Now we can look at the colData to see what was added:
+
+```{r view coldata stats}
+head(colData(filtered_sce))
+```
+
+We can also plot some of these statistics, here using the `plotMetrics()` function from the `miQC` package to plots the percent of reads that are mitochondrial (the `subsets_mito_percent` column) against the number of unique genes detected (the `detected` column) for each cell.
+
+```{r miQC plotMetrics}
+# use miQC::plotMetrics()
+miQC::plotMetrics(filtered_sce) + theme_bw()
+```
+
+We can see that there is a range of mitochondrial percentages, and it does also seem that genes with high percentages of mitochondrial cells don't seem to detect very many unique genes.
+
+One option is to define a cutoff for the mitochondrial percentage above which we call a cell compromised and exclude it from further analysis. 
```

Comment:
Add transition here, eg (or something else like this to remind everyone that we are filtering) - 
```suggestion
How do we filter with this information?
One option is to define a cutoff for the mitochondrial percentage above which we call a cell compromised and exclude it from further analysis. 
```

Reply:
^ lol I enjoy how the "we" is not a diff.